### PR TITLE
Deprecate get_shape() methods

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/common/low_precision_transformations/src/network_helper.cpp [hunk 1]
```diff
@@ -1560,10 +1560,12 @@ NetworkHelper::InsertDequantizationResult NetworkHelper::moveDequantizationBefor
     if (is_type<ov::opset1::Concat>(operation)) {
         const auto concatNode = as_type_ptr<ov::opset1::Concat>(operation);
         auto axis = concatNode->get_concatenation_axis();
-        if (dequantization.multiply && dequantization.multiplyConstant->get_shape().size() > 1 && dequantization.multiplyConstant->get_shape()[axis] != 1) {
+        if (dequantization.multiply && dequantization.multiplyConstant->get_output_partial_shape(0).to_shape().size() > 1 &&
+            dequantization.multiplyConstant->get_output_partial_shape(0).to_shape()[axis] != 1) {
             multiplyConstants = NetworkHelper::splitConstantsBeforeConcat(operation, { dequantization.multiplyConstant });
         }
-        if (dequantization.subtract && dequantization.subtractConstant->get_shape().size() > 1 && dequantization.subtractConstant->get_shape()[axis] != 1) {
+        if (dequantization.subtract && dequantization.subtractConstant->get_output_partial_shape(0).to_shape().size() > 1 &&
```
Comment (vurusovs) [inline src/common/low_precision_transformations/src/network_helper.cpp:1567]: may it be better to use `get_output_partial_shape(0).size()` directly?

File: src/common/snippets/src/op/scalar.cpp [hunk 2]
```diff
@@ -18,13 +18,15 @@ void Scalar::validate_and_infer_types() {
     Constant::validate_and_infer_types();
     auto out_pshape = get_output_partial_shape(0);
     NODE_VALIDATION_CHECK(this, out_pshape.is_static(), "Scalar supports only static input shapes");
-    NODE_VALIDATION_CHECK(this, out_pshape.get_shape().empty() || ov::shape_size(out_pshape.get_shape()) == 1,
-                      "Scalar supports only one-element constants, got ", out_pshape.get_shape(),
-                      " shape");
+    NODE_VALIDATION_CHECK(this,
+                          out_pshape.get_shape().empty() || ov::shape_size(out_pshape.get_shape()) == 1,
```
Comment (vurusovs) [inline src/common/snippets/src/op/scalar.cpp:22]: Is `get_shape()` used intentionally here?

Comment (ilyachur) [reply src/common/snippets/src/op/scalar.cpp:22]: to_shape and get_shape do the same things for partial shape

**Comment by ilya-lavrenov** [comment]
Decided trying to change existing declaration for this method to return by value instead of reference

**Pull request closed without merging**